stream-tests/src/test/scala/org/apache/pekko/stream/scaladsl/FlowMapWithResourceSpec.scala

```diff
@@ -0,0 +1,416 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.pekko.stream.scaladsl
+
+import java.io.BufferedReader
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.util.concurrent.atomic.AtomicInteger
+
+import scala.annotation.{ nowarn, tailrec }
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.{ Await, Promise }
+import scala.concurrent.duration.DurationInt
+import scala.util.Success
+import scala.util.control.NoStackTrace
+
+import com.google.common.jimfs.{ Configuration, Jimfs }
+import org.apache.pekko
+import pekko.Done
+import pekko.stream.{ AbruptTerminationException, ActorAttributes, ActorMaterializer, SystemMaterializer }
+import pekko.stream.ActorAttributes.supervisionStrategy
+import pekko.stream.Supervision.{ restartingDecider, resumingDecider }
+import pekko.stream.impl.{ PhasedFusingActorMaterializer, StreamSupervisor }
+import pekko.stream.impl.StreamSupervisor.Children
+import pekko.stream.testkit.{ StreamSpec, TestSubscriber }
+import pekko.stream.testkit.Utils.{ assertDispatcher, TE, UnboundedMailboxConfig }
+import pekko.stream.testkit.scaladsl.{ TestSink, TestSource }
+import pekko.testkit.EventFilter
+import pekko.util.ByteString
+
+class FlowMapWithResourceSpec extends StreamSpec(UnboundedMailboxConfig) {
+
+  private val fs = Jimfs.newFileSystem("MapWithResourceSpec", Configuration.unix())
+  private val ex = new Exception("TEST") with NoStackTrace
+
+  private val manyLines = {
+    ("a" * 100 + "\n") * 10 +
+    ("b" * 100 + "\n") * 10 +
+    ("c" * 100 + "\n") * 10 +
+    ("d" * 100 + "\n") * 10 +
+    ("e" * 100 + "\n") * 10 +
+    ("f" * 100 + "\n") * 10
+  }
```

Comment:
('a' to 'f')  .map(_.toString)  .map(_*100).map(_+"/n") ....etc

Reply:
Was copied from another test iirc, it's test code , and very explicitly, anyway, thanks